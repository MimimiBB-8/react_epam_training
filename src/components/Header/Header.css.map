{"version":3,"sourceRoot":"","sources":["Header.sass"],"names":[],"mappings":"AAAA;EACE;;;AAEF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;;;AACF;EACE;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;EACA;EACA;EACA;;AACA;EACE;EACA;;;AAEJ;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;EACA;EACA;;;AAEF;EACE","file":"Header.css"}