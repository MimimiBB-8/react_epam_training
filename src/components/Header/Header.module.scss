* {
  box-sizing: border-box;
}

.header {
  width: 1200px;
  position: relative;
  border: solid 2px #555555;
  background: url('../../img/header_background.png') 0px 0px no-repeat;
  background-size: cover;
  margin-bottom: 5px;
  z-index: 1;

  &:before {
    content: '';
    display: block;
    height: 101%;
    position: absolute;
    top: -1px;
    left: 0;
    right: -1px;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.8);
    z-index: -1;
  }
}

.content {
  //display: none;
  width: 100%;
  padding: 20px 60px;
  z-index: 3;
}

.first_line {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 35px;
}

.logo {
  width: 152px;
  height: 24px;
  font-weight: 500;
  font-size: 20px;
  line-height: 24px;
  color: #f65261;

  span {
    font-weight: 900;
    color: #f65261;
  }
}

h2 {
  font-size: 40px;
  font-weight: 300;
  line-height: 49px;
  letter-spacing: 1px;
  margin-bottom: 38px;
}

.search_content {
  margin-left: 60px;
}
