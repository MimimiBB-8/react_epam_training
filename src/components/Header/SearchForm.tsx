import React from "react";
import './Searchform.css'
import Button from "../Button/Button";

const SearchForm: React.FC = () => (
    <form>
        <input type={'text'} placeholder={'What do you want to watch?'}/>
        <Button title={'search'} classname={'search_button'}/>
    </form>
);

export default SearchForm;
