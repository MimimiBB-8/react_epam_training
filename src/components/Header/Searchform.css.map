{"version":3,"sourceRoot":"","sources":["Searchform.sass"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;;;AACF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA","file":"Searchform.css"}