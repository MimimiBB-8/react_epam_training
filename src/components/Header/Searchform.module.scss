form {
  display: flex;
  align-items: center;
  margin-bottom: 130px;
}

input {
  background: rgba(50, 50, 50, 0.8);
  border: none;
  border-radius: 4px;
  height: 57px;
  width: 713px;
  margin-right: 15px;
  font-weight: 400;
  font-size: 20px;
  line-height: 24px;
  color: #FFFFFF;
  padding-left: 19px;
}