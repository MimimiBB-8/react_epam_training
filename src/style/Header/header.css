* {
  box-sizing: border-box;
}

.header {
  height: 396px;
  width: 1200px;
  position: relative;
  border: solid 2px #555555;
  padding: 20px 60px;
  background-image: url("../../img/header_background.png");
  background-color: #232323;
  margin-bottom: 5px;
}

.header:before {
  position: absolute;
  content: "";
  display: block;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.8);
}

.content {
  position: absolute;
}

.first_line {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 35px;
}

.logo {
  width: 152px;
  height: 24px;
  font-weight: 900;
  font-size: 20px;
  line-height: 24px;
  color: #F65261;
}

.add_movie {
  background: rgba(96, 96, 96, 0.68);
  font-weight: 600;
  font-size: 20px;
  text-transform: uppercase;
  color: #F65261;
  padding: 11px 18px;
  height: 46px;
  width: 177px;
  border-radius: 4px;
}

h2 {
  font-size: 40px;
  font-weight: 300;
  line-height: 49px;
  letter-spacing: 1px;
  margin-bottom: 38px;
}

.search_content {
  margin-left: 60px;
}

/*# sourceMappingURL=header.css.map */
