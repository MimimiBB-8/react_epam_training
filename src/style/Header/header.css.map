{"version":3,"sourceRoot":"","sources":["header.sass"],"names":[],"mappings":"AAAA;EACE;;;AAEF;EACE;EACA;EAEA;EACA;EACA;EAEA;;;AAiBF;EACE;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA;EACA;EACA;;;AAEF;EACE","file":"header.css"}