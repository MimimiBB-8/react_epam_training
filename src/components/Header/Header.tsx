import style from './Header.module.scss'
import SearchForm from './SearchForm'

const Header = () => (
  <div className={style.header}>
    <div className={style.content}>
      <div className={style.first_line}>
        <label className={style.logo}>
          <span>netflix</span>roulette
        </label>
        <a className={style.add_movie}>+ ADD MOVIE</a>
      </div>
      <div className={style.search_content}>
        <h2>FIND YOUR MOVIE</h2>
        <SearchForm />
      </div>
    </div>
  </div>
)

export default Header
