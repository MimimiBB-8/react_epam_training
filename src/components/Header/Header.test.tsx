import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';


describe('component header', () => {
    it('render header component', () => {
        render(
            <Router>
                <Header />
            </Router>,
        );
        expect(screen.queryByRole('header')).toBeInTheDocument;
        expect(screen.queryByDisplayValue('netflixroulette')).toBeInTheDocument;
        expect(screen.queryByTitle('+ ADD MOVIE')).toBeInTheDocument;
        expect(screen.queryByText('FIND YOUR MOVIE')).toBeInTheDocument;
        expect(screen.queryByRole('form')).toBeInTheDocument;
    });
    it('snapshot header', () => {

        const { asFragment } = render(
            <Router>
                <Header />
            </Router>,
        );
        expect(asFragment()).toMatchSnapshot()
    })
})