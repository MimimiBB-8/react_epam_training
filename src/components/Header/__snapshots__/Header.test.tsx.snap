// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component header snapshot header 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="content"
    >
      <div
        class="first_line"
      >
        <label
          class="logo"
        >
          <span>
            netflix
          </span>
          roulette
        </label>
        <button
          class="add_movie"
        >
          + ADD MOVIE
        </button>
      </div>
      <div
        class="search_content"
      >
        <h2>
          FIND YOUR MOVIE
        </h2>
        <form
          aria-label="form"
          class="search_form"
        >
          <input
            class="search_form_input"
            name="search"
            placeholder="What do you want to watch?"
            title="search"
            type="search"
          />
          <button
            class="search_button"
            type="submit"
          >
            search
          </button>
        </form>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
